CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVReader.java

```diff
@@ -47,44 +62,171 @@ static void sendNotification(String filename, int numPacket, ArrayList<Map<Strin
+    static public ArrayList<Map<String, String>> executeFixedRecord(String csvFile, Map<String, Object> config,
+            ExtensionWebSocketClient oClient) throws InterruptedException {
+
+        int numOfRecords; // This is the total number of records/lines processed from the file.
+        int packetIndex = 0;
+        Map<String, Map<String, String>> schema = null;
+        Map<String, FixedRecordfieldInfo> recordMetaData = null;
+
+        if (config.get("schema") != null) {
+            schema = (Map<String, Map<String, String>>) config.get("schema");
+        }
+
+        recordMetaData = fixedRecord(schema);
+
+        int recordSize = fixedRecordLength(recordMetaData);
+        if (config.get("fixedRecordSize") != null)
+            recordSize = (int) config.get("fixedRecordSize");
+
+        int MaxLinesInEvent = (int) config.get(MAX_LINES_IN_EVENT);
+
+        int SleepBetweenPackets = 0;
+        if (config.get("waitBetweenTx") != null)
+            SleepBetweenPackets = (int) config.get("waitBetweenTx");
+
+        ArrayList<Map<String, String>> file = new ArrayList<Map<String, String>>();
+        Set<String> fieldList = recordMetaData.keySet();
+
+        try (InputStream inputStream = new FileInputStream(csvFile)) {
+            numOfRecords = 0;
+            byte[] tempBuffer = new byte[recordSize];
+            while (inputStream.read(tempBuffer) != -1) {
+                Map<String, String> lineValues = new HashMap<String, String>();
+
+                for (String key : fieldList) {
+
+                    FixedRecordfieldInfo o = recordMetaData.get(key);
+                    String t;
+                    if (o.charSet != null) {
+                        t = new String(tempBuffer, o.offset, o.length, o.charSet).trim();
+                    } else {
+                        t = new String(tempBuffer, o.offset, o.length).trim();
+                    }
+
+                    if (o.reversed) {
+                        StringBuilder sb = new StringBuilder();
+                        sb.append(t);
+                        sb = sb.reverse();
+                        t = sb.toString();
+                    }
+
+                    lineValues.put(key, t);
+                }
+
+                file.add(lineValues);
+
+                numOfRecords++;
+
+                if (file.size() >= MaxLinesInEvent) {
+                    log.info("TX Packet {} Size {} Total num of Records {}", packetIndex, MaxLinesInEvent,
+                            numOfRecords);
+                    sendNotification(csvFile, packetIndex, file, oClient);
+                    Thread.sleep(SleepBetweenPackets);
+                    file = new ArrayList<Map<String, String>>();
+                    packetIndex++;
+                }
+
+            }
+            if (file.size() > 0) {
+                log.info("TX Last Packet Packet {} Size {} Total num of Records {}", packetIndex, MaxLinesInEvent,
```

Comment:
Isn't doing these as `.info()`'s going to dump a bunch of stuff into the log file if this is really heavily used?  Seems like this is a `.debug()` type thing.

If you must see progress, maybe do dump file-level statistics.

Reply:
i still needs it, its not to much info,i guess when it would be more stable i will add a debug feature which will dump it .